### PR TITLE
Boiler Tweaks

### PR DESCRIPTION
**Pull request by Barnet2**
## About The Pull Request
Basically puts boiler alot closer to CM boiler now that we are starting to take down offscreens.
Zoom reduced to about miniscope range (4 tiles)
Opaque gas is back
Boiler gas radius reduced from 4 to 3 tile radius
Able to shoot within 5 tiles of yourself
## Why It's Good For The Game
Boiler was a lot less toxic without opaque gas but did lack the utility a smokescreen gave. The toxicity was mostly because you literally could not interact with boiler at sniper range, even with root. This makes boiler have a good more deal risk/reward as they are much, much closer to marines but have the smokescreen. Additionally no zoom windup as that felt quite arbitrary.
## Changelog
:cl:
balance: Boiler - Zoom reduced to about miniscope range (4 tiles)
balance: Boiler - Opaque gas is back
balance: Boiler - Boiler gas radius reduced from 4 to 3 tile radius
balance: Boiler - Able to shoot within 5 tiles of yourself
/:cl:

### CODE REVIEW

**Comment by MiniMeatwad** [comment]
One day TGMC will remove every way to attack people from off-screen and have the game be based on on-screen encounters.

As Rohesie intended.

**Comment by Clay500z** [comment]
I think you should also either remove or reduce the minimum tile distance (whatever it's called) for boiler to shoot. Having to click on the very edge of your screen to shoot without zoom, I mean. It's mostly just an inconvenience, but sometimes you want to shoot yourself and there's no walls around.

**Comment by MLGTASTICa** [comment]
surprise surprise yet another crusher buff. When will the boiler nerfs stop? Do we really need to force everyone to go crusher to siege a fob now ? who expects a crusher with 4 tile zoom to be able to siege when the sniper rifles now outrange it 3 fold ???

**Comment by LohnRan** [comment]
> surprise surprise yet another crusher buff. When will the boiler nerfs stop? Do we really need to force everyone to go crusher to siege a fob now ? who expects a crusher with 4 tile zoom to be able to siege when the sniper rifles now outrange it 3 fold ???

https://github.com/tgstation/TerraGov-Marine-Corps/pull/15998#issue-2332343046

**Comment by Lefunny** [comment]
Doing that without giving boiler betters way to run or increasing his damage is a bad idea, jetpacks can probably insta reach a boiler with said change too

**Comment by SimplySmiley123** [comment]
> Doing that without giving boiler betters way to run or increasing his damage is a bad idea, jetpacks can probably insta reach a boiler with said change too

The marksman weapons are getting their numbers limited and scopes are getting axed.
If anything boiler will deserve this with those changes.

**Comment by Lefunny** [comment]
> > Doing that without giving boiler betters way to run or increasing his damage is a bad idea, jetpacks can probably insta reach a boiler with said change too
> 
> The marksman weapons are getting their numbers limited and scopes are getting axed. If anything boiler will deserve this with those changes.

It's a schizophrenic nerf on marksmen reducing their number to four, i don't even see four snipers in a game only alone will fuck boiler

**Comment by SimplySmiley123** [comment]
> > > Doing that without giving boiler betters way to run or increasing his damage is a bad idea, jetpacks can probably insta reach a boiler with said change too
> > 
> > 
> > The marksman weapons are getting their numbers limited and scopes are getting axed. If anything boiler will deserve this with those changes.
> 
> It's a schizophrenic nerf on marksmen reducing their number to four, i don't even see four snipers in a game only alone will fuck boiler

Nope. One marksman weapon barely puts much pressure on a boiler, especially when the boiler doesn't need to commit to his placement at all.
AMR? Sure, that puts some pressure.
Regular bog standard marksman weapons? Lol lmao, not without a firing squad of them.

**Comment by Barnet2** [comment]
This PR is partially because I think rushing/movement is a good deal more interactive for everyone as a whole instead of a sniper and boiler trying to duel 3 screens away from eachother. I doubt a good boiler will get rushed too much (they can definitely get pressured now) but if it gets too bad I can definitely port Acid Shroud from CM, which is basically just a glob ontop of yourself to GTFO.

**Comment by hivemuscle** [comment]
pls remove 3 seconds long dig-in so i can finally play cqc boiler and have fun

**Comment by Arturlang** [comment]
You might have to make the boiler a bit more tanky because this means any marine with explosives can run up to the boiler and instagib it. 0 armor moment

**Comment by Barnet2** [comment]
> You might have to make the boiler a bit more tanky because this means any marine with explosives can run up to the boiler and instagib it. 0 armor moment

Definitely a possibility, we'll see though

**Comment by AusarAtkin** [comment]
> pls remove 3 seconds long dig-in so i can finally play cqc boiler and have fun, i fucking love cqc boiler

The 3 seconds dig in been a part of cqc boiler challenges ever since boiler existed.

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 1]
```diff
@@ -33,6 +33,12 @@ GLOBAL_LIST_INIT(boiler_glob_image_list, list(
 	keybinding_signals = list(
 		KEYBINDING_NORMAL = COMSIG_XENOABILITY_LONG_RANGE_SIGHT,
 	)
+	var/tile_offset = 5
+	var/view_size = 4
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:37]: autodoc

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 2]
```diff
@@ -43,9 +49,9 @@ GLOBAL_LIST_INIT(boiler_glob_image_list, list(
 	else
 		X.visible_message(span_notice("[X] starts looking off into the distance."), \
 			span_notice("We start focusing your sight to look off into the distance."), null, 5)
-		if(!do_after(X, 1 SECONDS, IGNORE_HELD_ITEM, null, BUSY_ICON_GENERIC) || (X.xeno_flags & XENO_ZOOMED))
+		if(X.xeno_flags & XENO_ZOOMED)
 			return
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:53]: ? this can never be true.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Review by Lumipharon — APPROVED** [review]

**Comment by Kuro020** [comment]
although the scope pr is dead, xenos got pretty strong now with buffs, so let's test this

**Comment by Technical-l** [comment]
> > This PR is partially because I think rushing/movement is a good deal more interactive for everyone as a whole instead of a sniper and boiler trying to duel 3 screens away from eachother. I doubt a good boiler will get rushed too much (they can definitely get pressured now) but if it gets too bad I can definitely port Acid Shroud from CM, which is basically just a glob ontop of yourself to GTFO.
> 
> what's to stop a marine from rushing boiler with mimir 2? boiler has no survivability

other fucking xenos?

**Comment by Ephemeralis** [comment]
> although the scope pr is dead, xenos got pretty strong now with buffs, so let's test this

Who could've seen this coming?

**Comment by Clay500z** [comment]
Some observations I made about this new Boiler:

1. Full glow Boiler is the only viable way to play now. You'll usually be in range of flares or ontop of acid wells anyways, so no glow has very little point to it.

2. You can zoom and prepare a bombard at the same time now. Don't have to make the choice between zooming or preparing bombard first.

3. Mg27 minscope is now the hard counter to Boilers, an even harder counter than snipers. It kills you faster than snipers, can suppress or destroy cover you're hiding behind, and has a bagillion bullets so missing isn't an issue. All while still shooting at any other xenos nearby whenever you aren't peeking, because running out of ammo in the mag when you peek is less of an issue for it. Only advantage snipers have over it is that snipers can cover a larger area and more flanks at the same time.

4. A single marine with a miniscope can stop you from shooting out of cover at all. A rusher with a shotgun will also stagger and interrupt your bombard very easily. That is to say, you'll need to shoot from behind cover 75% of the time. Addendum to point 1, no glow let's you shoot out of cover more often if there's darkness, but I'd say it's not worth the massive cooldown bombard gets. And it relies on marines not using flares. Or fire. Or blindly shooting into the darkness. Or wandering a little too close to you. Or Baldur.

5. A Boiler is significantly stronger with one other Boiler or Defiler. Wait for the other one to drop its gas, and then simply shoot your bombard while inside it's gas. Solves the problem of marines rushing or targeting you when you're out of cover, cause they just can't see you.


Overall a decent change, but it needs some tweaking to make it less cbt to play. I'd say to either give it stagger immunity or increase the zoom range by another 5 tiles. It's far too easy for bombard to get interrupted by shotgun staggers now, stagger immunity should let the boiler shoot out of cover with the risk of dying to bullets. Alternatively an increased zoom will just keep it outside of stagger range, but that brings back it's off-screening.

**Comment by Barnet2** [comment]
> Some observations I made about this new Boiler:
> 
>     1. Full glow Boiler is the only viable way to play now. You'll usually be in range of flares or ontop of acid wells anyways, so no glow has very little point to it.
> 
>     2. You can zoom and prepare a bombard at the same time now. Don't have to make the choice between zooming or preparing bombard first.
> 
>     3. Mg27 minscope is now the hard counter to Boilers, an even harder counter than snipers. It kills you faster than snipers, can suppress or destroy cover you're hiding behind, and has a bagillion bullets so missing isn't an issue. All while still shooting at any other xenos nearby whenever you aren't peeking, because running out of ammo in the mag when you peek is less of an issue for it. Only advantage snipers have over it is that snipers can cover a larger area and more flanks at the same time.
> 
>     4. A single marine with a miniscope can stop you from shooting out of cover at all. A rusher with a shotgun will also stagger and interrupt your bombard very easily. That is to say, you'll need to shoot from behind cover 75% of the time. Addendum to point 1, no glow let's you shoot out of cover more often if there's darkness, but I'd say it's not worth the massive cooldown bombard gets. And it relies on marines not using flares. Or fire. Or blindly shooting into the darkness. Or wandering a little too close to you. Or Baldur.
> 
>     5. A Boiler is significantly stronger with one other Boiler or Defiler. Wait for the other one to drop its gas, and then simply shoot your bombard while inside it's gas. Solves the problem of marines rushing or targeting you when you're out of cover, cause they just can't see you.
> 
> 
> Overall a decent change, but it needs some tweaking to make it less cbt to play. I'd say to either give it stagger immunity or increase the zoom range by another 5 tiles. It's far too easy for bombard to get interrupted by shotgun staggers now, stagger immunity should let the boiler shoot out of cover with the risk of dying to bullets. Alternatively an increased zoom will just keep it outside of stagger range, but that brings back its off-screening.
> 
> Also please remove the minimum range for bombarding. Having to click on the very edge of the screen to shoot is just anti-qol without any real balance implication.

Stagger and closer range weapons are infact supposed to prevent you from your very high-impact gas. I didn't have an issue with miniscope guns as you can get gas off a tiny bit out of their range, which promotes them moving instead of holding down M1. I agree that the glob system probably doesn't have a place anymore but I am hesitant to remove it as it does give a bit of depth to the caste. 

Re-added the zoom timer and removed the minimum distance to fire. Additionally on other feedback and my observations, reduced gas radius back to what it was before root removal.

**Comment by Fawcksy** [comment]
Very interesting rebalance in theory, seems to be working in practice. idk what else to say other than you get le epic upboat

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**